src/swap/UniswapV3SwapAdapter.sol

```diff
@@ -17,7 +17,7 @@ import { IV3SwapRouter } from "./interfaces/uniswap/IV3SwapRouter.sol";
  *         MetaStreet Foundation
  *         Adapted from https://github.com/metastreet-labs/metastreet-usdai-contracts/blob/main/src/swapAdapters/UniswapV3SwapAdapter.sol
  */
-contract UniswapV3SwapAdapter is IUniswapV3SwapAdapter, AccessControl, ReentrancyLock {
+contract UniswapV3SwapAdapter is IUniswapV3SwapAdapter, ReentrancyLock, AccessControlUpgradeable {
     using SafeERC20 for IERC20;
```

Comment:
Do we want this contract to be upgradeable?
In the README it mentions that this contract is immutable:
https://github.com/m0-foundation/m-extensions/blob/e96e0b16baac02416f1103895e8cce437f3953c0/README.md?plain=1#L62